lib/Service/QuickActionsService.php

```diff
@@ -35,13 +35,18 @@ public function __construct(
 		private ActionStepMapper $actionStepMapper,
 	) {
 	}
-
 	/**
 	 * @param string $userId
 	 * @return Actions[]
 	 */
 	public function findAll(string $userId): array {
-		return $this->actionsMapper->findAll($userId);
+		$actions = $this->actionsMapper->findAll($userId);
+		foreach ($actions as $action) {
+			$actionSteps = $this->actionStepMapper->findAllStepsForOneAction($action->getId(), $userId);
```

Comment:
perf: *n+1 problem*

you could fetch the steps in bulk with a WHERE action_id IN (...)